### ISSUE TITLE
Ignoring change of runtime

### ISSUE THREAD

**Issue by pedroaugustosmribeiro** [microsoft/vscode-data-wrangler#358]
Type: <b>Bug</b>

Can't change runtime of Data Wrangler, it ignores my choice and tries to use the last used one, even after clearing cache

# Diagnostic data

-   Jupyter extension version: 2024.11.0
-   Python extension version: 2024.23.2024121701
-   .NET Install Tool for Extension Authors extension version: 2.2.3
-   Python package dependencies:

```json
{
  "installed": {
    "pyarrow": "16.0.0",
    "openpyxl": "3.1.2"
  },
  "required": {
    "pandas": "1.2.0",
    "openpyxl": "0.0.1"
  },
  "unsatisfied": [
    "pandas"
  ]
}
```

-   Entrypoint: Reopen
-   Active mode: dataWrangler

Extension version: 1.14.0
VS Code version: Code 1.96.0 (138f619c86f1199955d53b4166bef66ef252935c, 2024-12-11T02:29:09.626Z)
OS version: Windows_NT x64 10.0.22631
Modes:
Remote OS version: Linux x64 6.5.0-41-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 9 7900X 12-Core Processor             (24 x 4691)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|31.59GB (18.41GB free)|
|Process Argv|--crash-reporter-id 53481740-fa21-443e-84e5-a7d0b0d9de7a|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|SSH: ***|
|OS|Linux x64 6.5.0-41-generic|
|CPUs|AMD Ryzen Threadripper PRO 5995WX 64-Cores (128 x 1800)|
|Memory (System)|755.32GB (734.12GB free)|
|VM|0%|
</details>

**Comment by pwang347** [comment]
Hi @pedroaugustosmribeiro, thanks for opening this issue.
I tried with the extension versions you've provided, but I do not seem to run into this problem. Are you changing the kernel using the kernel selector below? If so, is it showing the old kernel name here for you even after switching?

<img width="110" alt="Image" src="https://github.com/user-attachments/assets/52f54ef5-bb0d-4162-b6f8-e3e220ab7ebd" />
<img width="141" alt="Image" src="https://github.com/user-attachments/assets/75caffcf-1f92-4f1b-a06d-2b317be71d3c" />

Also, when you say you tried clearing cache, did you mean you ran this command (accessed via ctrl+shift+P)?

<img width="605" alt="Image" src="https://github.com/user-attachments/assets/cf13d015-af2f-4c2c-91cb-054c71b3b6fb" />


Thanks!

**Comment by pedroaugustosmribeiro** [comment]
Yes, it shows the environment selected, and yes, I have run that command, with no success on a new xslx file that Data Wrangler never opened before. When trying with an older file previously oppend, it worked correctly with the changed environment

**Comment by pwang347** [comment]
@pedroaugustosmribeiro Actually, it seems like the behaviour for kernel change is incorrect.

Running the following code, it reveals that although the kernel name is changing the underlying kernel path is still using the original one:
```python
import sys
df = pd.DataFrame([sys.executable])
```

I will look into fixing this issue and let you know when the fix is available. Thanks!

**Comment by pwang347** [comment]
Hi @pedroaugustosmribeiro, thank you for your patience. Just wanted to share that this issue should be addressed in the latest pre-release version 1.17.0. Please give it a try and let me know if you encounter any issues with it. Hope this helps!

**Comment by pedroaugustosmribeiro** [comment]
Hi, sorry for the wait.
I tested version 1.17.1 and it seems there's an issue with long env names:

```py
19:15:48.421 [info] Process Execution: ~/.conda/envs/miaoper_23.10.1_extension_~_24.11/bin/python -m 
```
should be `~/.conda/envs/miaoper_23.10.1_extension_pedro_24.11/bin/python`

**Comment by pwang347** [comment]
Hi @pedroaugustosmribeiro, no worries and thank you for the reply!

Can you please confirm the following:
* Changing the runtime still does not work in 1.17.1 (the long env name is not a separate issue)
* If so, can you please share the exact steps you are taking to reproduce the issue? For example, 1. run `Data Wrangler: Clear cached runtime` 2. Reload VS Code 3. Open CSV file 4. Run `import sys; df = pd.DataFrame([sys.executable])` in Data Wrangler 5. Observe that the runtime is incorrect
* Are you able to use/change the runtime in a Jupyter notebook (.ipynb) file? We are using the Jupyter extension's API to access the environment, and it would be great if you can confirm that this part is working.

Thanks!

**Comment by pedroaugustosmribeiro** [comment]
> Hi [@pedroaugustosmribeiro](https://github.com/pedroaugustosmribeiro), no worries and thank you for the reply!
> 
> Can you please confirm the following:
> 
> * Changing the runtime still does not work in 1.17.1 (the long env name is not a separate issue)
> * If so, can you please share the exact steps you are taking to reproduce the issue? For example, 1. run `Data Wrangler: Clear cached runtime` 2. Reload VS Code 3. Open CSV file 4. Select the env with the long name 5. Run `import sys; df = pd.DataFrame([sys.executable])` in Data Wrangler 6. Observe that the runtime is incorrect
> * Are you able to use/change the runtime in a Jupyter notebook (.ipynb) file? We are using the Jupyter extension's API to access the environment, and it would be great if you can confirm that this part is working.
> 
> Thanks!

Hi @pwang347 
I could only complete steps 1-4, the same error at the log as before.
On Jupyter Notebook, the same problem when changing runtime to the long name env

**Comment by pedroaugustosmribeiro** [comment]
Working now, thank you

**Comment by pwang347** [comment]
@pedroaugustosmribeiro glad you were able to resolve the issue, will close this for now. If the issue arises again, based on your description I would recommend opening an issue on the Jupyter extension if it is also affecting them (since we are using their kernels) and they may be able to help resolve it on their end. Cheers!
